backend/src/models/reports/Tab_37_rpt_PF_BudgetDistributionbyPortfolio.js

```diff
@@ -1,16 +1,216 @@
+          Fiscal_Year fy 
+          right join Project_Deliverable pd on fy.ID = pd.Fiscal
+        ) 
+        left join ${baseQueries.q1} on (pd.Fiscal = q1.Fiscal) 
+        and (pd.Project_ID = q1.Project_ID)
+      ) 
+      right join (
+        Portfolio po 
+        right join Project_Budget pb ON po.ID = pb.Recovery_Area
+      ) on pd.ID = pb.Project_Deliverable_ID
+    ) ON p.ID = pd.Project_ID 
+  group by 
+    p.ID, 
+    p.Project_Number, 
+    p.Project_Name, 
+    po.Portfolio_Name, 
+    po.Portfolio_Abbrev, 
+    p.Total_Project_Budget, 
+    p.Recoverable_Amount, 
+    q1.Fees, 
+    q1.Expenses, 
+    q1.Total_Contract, 
+    pd.Fiscal, 
+    fy.Fiscal_Year 
+  having 
+    pd.fiscal = ${fiscal}
+    )  AS q2`
+    ),
+  q3: (fiscal) =>
+    knex.raw(
+      `(
+      select 
+    q2.Project_Number, 
+    q2.Project_Name, 
+    p.Recoverable, 
+    q2.Total_Project_Budget, 
+    q2.Recoverable_Amount, 
+    q2.Total_Contract, 
+    q2.Fiscal_Year, 
+    Sum(
+      case when q2.Portfolio_Abbrev = 'OSS' then q2.Current_FY_Total_Recoverable else null end
+    ) AS OSS, 
+    Sum(
+      case when q2.Portfolio_Abbrev = 'DES' then q2.Current_FY_Total_Recoverable else null end
+    ) AS DES, 
+    Sum(
+      case when q2.Portfolio_Abbrev = 'DMS' then q2.Current_FY_Total_Recoverable else null end
+    ) AS DMS, 
+    Sum(
+      case when q2.Portfolio_Abbrev = 'DP' then q2.Current_FY_Total_Recoverable else null end
+    ) AS DP, 
+    Sum(
+      case when q2.Portfolio_Abbrev = 'ANA' then q2.Current_FY_Total_Recoverable else null end
+    ) AS ANA, 
+    Sum(
+      case when q2.Portfolio_Abbrev = 'SD' then q2.Current_FY_Total_Recoverable else null end
+    ) AS SD, 
+    Sum(
+      case when q2.Portfolio_Abbrev = 'CE' then q2.Current_FY_Total_Recoverable else null end
+    ) AS CE, 
+    Sum(
+      case when q2.Portfolio_Abbrev = 'EDS' then q2.Current_FY_Total_Recoverable else null end
+    ) AS EDS, 
+    Sum(
+      case when q2.Portfolio_Abbrev = 'BCS' then q2.Current_FY_Total_Recoverable else null end
+    ) AS BCS, 
+    Sum(
+      case when q2.Portfolio_Abbrev = 'DIV' then q2.Current_FY_Total_Recoverable else null end
+    ) AS DIV, 
+    Sum(
+      q2.Current_FY_Total_Recoverable
+    ) as Current_FY_Recoverable_Total 
+  from 
+    ${baseQueries.q2(fiscal)} 
+    inner join Project p on q2.Project_ID = p.ID 
+  group by 
+    q2.Project_Number, 
+    q2.Project_Name, 
+    p.Recoverable, 
+    q2.Total_Project_Budget, 
+    q2.Recoverable_Amount, 
+    q2.Total_Contract, 
+    q2.Fiscal_Year
+    ) AS q3`
+    ),
+};
+
+/**
+ * Retrieves the final queries for invoice payments, filtered by fiscal year
+ *
+ * @param   {number | string | Array} fiscal- The fiscal year to grab data for
+ * @returns {Promise}                         - A promise that resolves to the query result
+ */
+const reportQueries = {
+  fiscalYear: (fiscal) =>
+    knex("fiscal_year").select("fiscal_year").where("fiscal_year.id", fiscal).first(),
```

Comment:
i think in this case how you have done this fiscal does get automatically escaped, but we should verify as well

Reply:
This should be fine since it's inside a Knex `where()` function. We just need to avoid inserting variables that come from user input directly into the SQL string.

Reply:
Yes, knex queries are safe, though raw queries need to be escaped as above.